src/resources/Resource.js

```diff
@@ -11,23 +12,88 @@ export default class Resource {
     return url
   }
 
-  request (spec) {
-    let options = {
-      method: (spec.method || 'GET').toUpperCase()
+  async request ({ path, fields = [], params = {}, method = 'get' }) {
+    const options = {
+      method: method.toLowerCase(),
+      headers: {}
     }
-    let url = `${this.apiUrl}${spec.path}?__resellerID=10000&__i18n="en-gb"`
 
-    if (spec.fields) {
-      url += `&__fields=[${spec.fields.join(',')}]`
+    const args = {
+      __resellerID: this.api.resellerId,
+      __i18n: this.api.lang,
+      __fields: fields,
+      ...params
     }
+    const queryStr = formatArgs(args, options.method === 'post')
 
-    // Generate query string
-    if (spec.params) {
-      for (let param in spec.params) {
-        url += `&${param}=${spec.params[param]}`
+    let url
+    if (options.method === 'post') {
+      options.headers['Content-Type'] = 'application/x-www-form-urlencoded;charset=UTF-8'
+      options.body = queryStr
+      url = `${this.apiUrl}/${path.replace(/^\/+|\/+$/, '')}`
```

Comment:
This is how the API expects the request, so this is how I prepare the request...